server/routes/discover.ts

```diff
@@ -202,23 +253,25 @@ discoverRoutes.get('/tv', async (req, res) => {
   });
 });
 
-discoverRoutes.get<{ genreId: string }>(
-  '/tv/genre/:genreId',
-  async (req, res) => {
+discoverRoutes.get<{ language: string }>(
+  '/tv/language/:language',
+  async (req, res, next) => {
     const tmdb = createTmdbWithRegionLanaguage(req.user);
 
-    const genres = await tmdb.getTvGenres({
-      language: req.query.language as string,
-    });
+    const languages = await tmdb.getLanguages();
 
-    const genre = genres.find(
-      (genre) => genre.id === Number(req.params.genreId)
+    const language = languages.find(
+      (lang) => lang.iso_639_1 === req.params.language
     );
 
+    if (!language) {
+      return next({ status: 404, message: 'Unable to retrieve language' });
+    }
+
     const data = await tmdb.getDiscoverTv({
       page: Number(req.query.page),
       language: req.query.language as string,
-      genre: Number(req.params.genreId),
+      originalLanguage: req.params.language as string,
```

Comment:
Not sure why we need this type assertion here?

Reply:
Ah you're right, whoops!  Removed from both places. 😺